docs/arch/adr-003-versioning-branching-strategy.md

```diff
@@ -0,0 +1,66 @@
+# Release Versioning / Branching Strategy
+
+## Context
+With `dbt` ever evolving and progressing, we continue to ship and release versions on a regular basis. As we release `dbt`, versioning and branching become key to identifying what changes for users and conveying the expectations of those changes. We currently push all our changes to `main` which is the latest version of our code. We don't always want to release the latest version of our code though. We want to be able to isolate changes to go into particular releases based on urgency (e.g. bug fixes) and stability (e.g. new features). 
+
+We already have an established versioning strategy when it comes to major/minor/patch releases. The following is our approach as to what to expect from each release version:
+
+* Major: This signifies that breaking changes which require user action are included in this release.
+* Minor version: This signifies that new features are included in this release.
+* Patch version: This signifies that bug and security fixes are included in this release.
+
+We need a versioning strategy that allows us to to convey to users the confidence and stability of code they are installing. We want a versioning strategy that allows users to try out new changes, and experiment with them to get early feedback. We want to make sure users can also get a preview of new features we are working on, so they can incorporate these changes into their work as soon as possible. 
+
+We need a branching strategy to support the goals of the versioning strategy. We do not want to have to stop development on `main` as we await a release, nor do we want every change in `main` published in every release. Using branches, we can decide which changes go into which release.
+
+## Requirements
+The following are requirements that must be met for releasing dbt:
+
+1. We must have major, minor, and patch releases.
+1. We must have release types that have the goal of garnering early feedback from the Community.
+1. We must have release types that aim to identify bugs and test for stability.
```

Comment:
What is a release type

Reply:
I was struggling for a good word here. I can give examples (beta, rc, etc.). Any thought on a better term? I didn't want to use version since we are using version in the context of major, minor, and patch.

Reply:
release "stage" maybe?

Reply:
Ok updating to that. Thanks for the idea!

Reply:
[PEP 440](https://peps.python.org/pep-0440/#pre-releases) uses the term "phase" to describe options like beta, rc, etc within the pre-release "segment".

Reply:
I also like "phase."